Add default typographies

thomasdax98 opened the pull request:
See the result here: https://deploy-preview-1923--comet-admin.netlify.app/?path=/story/comet-admin-typography--typography

As defined by UX:

<img width="1062" alt="Bildschirmfoto 2024-04-08 um 13 38 01" src="https://github.com/vivid-planet/comet/assets/13380047/402e3376-c3a2-4361-8c95-403352e3407c">


---

<!-- Everything below this line will be removed from the commit message when the PR is merged -->

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
- [x] Task COM-575

thomasdax98 commented on .changeset/curvy-moles-punch.md:
```diff
@@ -5,4 +5,4 @@
 Rework `typographyOptions`
 
 - Replace `typographyOptions` with `createTypographyOptions()` to enable using the theme's breakpoints for media queries
-- Add new styles for `button` variant
```
This was removed again in the previous PR but I forgot to remove it from the changeset

johnnyomair commented on packages/admin/admin-theme/src/typographyOptions.ts:
```diff
@@ -9,6 +9,7 @@ export const fontWeights = {
     fontWeightRegular: 300,
     fontWeightMedium: 400,
     fontWeightBold: 500,
+    fontWeightVeryBold: 600,
```
Should we align the fontWeights with a common name mapping? https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#common_weight_name_mapping

jamesricky replied:
Generally I think we should, the problem is, every project would have to change every font-weight used from the theme. Maybe it's possible with a codemod, if not, that would be a lot of work to restore the previous style.

thomasdax98 replied:
Done a129b63d0fb49b3b3e4e5a5d6d0a1527583fc2d1 

@jamesricky I don't think it's used regularly. But we can make a codemod

thomasdax98 commented on packages/admin/admin-theme/src/typographyOptions.ts:
```diff
@@ -3,12 +3,18 @@ import { TypographyOptions } from "@mui/material/styles/createTypography";
-    fontWeightRegular: 300,
-    fontWeightMedium: 400,
-    fontWeightBold: 500,
+    thin: 100,
+    extraLight: 200,
+    light: 300,
+    regular: 400,
+    medium: 500,
+    semiBold: 600,
+    bold: 700,
+    extraBold: 800,
+    black: 900,
+    extraBlack: 950,
```
@jamesricky Thanks for the heads-up

I removed `extraBold`, `black` and `extraBlack` as I couldn't find them in the designs.

The new font weights can be used out-of-the-box because the system is based on simple pattern matching:

```tsx
<Typography fontWeight="thin">Thin</Typography>
<Typography fontWeight="extraLight">Extra Light</Typography>
<Typography fontWeight="light">Light</Typography>
<Typography fontWeight="regular">Regular</Typography>
<Typography fontWeight="medium">Medium</Typography>
<Typography fontWeight="semiBold">Semi Bold</Typography>
<Typography fontWeight="bold">Bold</Typography>
```

<img width="122" alt="Bildschirmfoto 2024-04-09 um 16 09 07" src="https://github.com/vivid-planet/comet/assets/13380047/f330f66d-cd66-4966-a39e-c84b1924a7e2">

There are no explicit types for the fontWeights (so there is also no type that must be extended).

---

I encountered a problem with the fontsource imports because it seems like the package doesn't have all weights. These are the files from the package:

<img width="181" alt="Bildschirmfoto 2024-04-09 um 15 45 49" src="https://github.com/vivid-planet/comet/assets/13380047/a62690f8-d925-4c35-9321-0201d0cb4390">

200 and 600 seem to be missing although we need them. Am I doing something wrong or is that okay?

jamesricky replied:
> 200 and 600 seem to be missing although we need them. Am I doing something wrong or is that okay?

That means we are not actually loading/using that font... seems like Roboto doesn't have the those weight's in general 🤔

thomasdax98 replied:
> That means we are not actually loading/using that font... seems like Roboto doesn't have the those weight's in general 🤔

How much of a problem is that? Doesn't the browser handle that as a fallback by just making it bolder?

jamesricky replied:
> Doesn't the browser handle that as a fallback by just making it bolder?

I think it does, I'm not sure how reliable/accurate this is. 
Maybe it just happens to look fine on our machines 😅 

But if UX explicitly defines a text to be rendered as Roboto 200, we should also do that instead of letting the browser guess how it should look. 

I asked UX where they got the font from, assuming they actually have it, maybe we can use that.

thomasdax98 commented:
@johnnyomair @jamesricky I replaced Roboto with Roboto Flex and also added a codemod for the update: https://github.com/vivid-planet/comet-upgrade/pull/11

jamesricky commented on demo/admin/package.json:
```diff
@@ -33,7 +33,7 @@
         "@emotion/react": "^11.9.3",
         "@emotion/styled": "^11.9.3",
-        "@fontsource/roboto": "^4.5.5",
+        "@fontsource-variable/roboto-flex": "^5.0.14",
```
This should be mentioned in the changeset, that the font also needs to be changed in the application.

jamesricky replied:
And/or mention the codemod added in https://github.com/vivid-planet/comet-upgrade/pull/11 (assuming it's not executed automatically)

thomasdax98 replied:
True, I forgot that. Done in 0f7136a561dc6d4699478a803c886e2fd68b7b2a
